pkg/blockstorage/azure/azuredisk.go

```diff
@@ -225,7 +225,12 @@ func (s *AdStorage) SnapshotCopyWithArgs(ctx context.Context, from blockstorage.
 		},
 	}
 
-	result, err := s.azCli.SnapshotsClient.CreateOrUpdate(ctx, s.azCli.ResourceGroup, snapName, createSnap)
+	migrateResourceGroup := s.azCli.ResourceGroup
+	if val, ok := args[blockstorage.AzureMigrateStorageKey]; ok && val != "" {
```

Comment:
This should be AzureMigrateResourceGroup instead of AzureMigrateStorageKey ?

Reply:
you are very correct